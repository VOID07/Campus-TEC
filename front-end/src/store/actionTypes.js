// constantes contra las que se comparan los valores al evaluar un reducer
export const SET_USER_PASSWORD = "SET_USER_PASSWORD";
export const SET_PROFILE = "SET_PROFILE";
export const SET_PROFILE_PROFESOR = "SET_PROFILE_PROFESOR";
