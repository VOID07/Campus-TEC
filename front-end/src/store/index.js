import { createStore, combineReducers } from "redux";
import login from "./tipo_usuario/reducer";

// Combina los diferentes reducers en uno solo para accederlos desde el store
const reducers = combineReducers({
  login,
});

// Store de todos los reducers
const store = createStore(reducers);

export default store;
