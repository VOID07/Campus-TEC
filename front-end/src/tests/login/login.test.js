import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import {cleanup, fireEvent, render} from '@testing-library/react';

import Login from "./../../componets/login/index";
import store from "./../../store/index";

//   let usrType = 1;
//   // manually trigger the callback
//   tree.props.setUserType(usrType)
//   // re-rendering
//   tree = component.toJSON();
//   expect(tree).toMatchSnapshot();

//   // manually trigger the callback
//   tree.props.onMouseLeave();
//   // re-rendering
//   tree = component.toJSON();
//   expect(tree).toMatchSnapshot();

test("Verificar la ausencia de cambios inesperados en el GUI", () => {
  const component = renderer.create(
    <Provider store={store}>
      <Login />
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.children[1].children[0].children[1].children[0].children[2].children[0].children[1].children[0].props.onChange();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.children[1].children[0].children[1].children[0].children[2].children[0].children[2].children[0].props.onChange();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.children[1].children[0].children[1].children[0].children[2].children[0].children[0].children[0].props.onChange();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

afterEach(cleanup);

it('Actualizar solo 1 elemento en checkbox', () => {
  const {queryByLabelText, getByLabelText} = render(
    <Provider store={store}>
      <Login />
    </Provider>
  );
expect(queryByLabelText("Estudiante").checked).toBeTruthy();
expect(queryByLabelText("Profesor").checked).toBeFalsy();
expect(queryByLabelText("Administrador").checked).toBeFalsy();

fireEvent.click(getByLabelText("Profesor"));

expect(queryByLabelText("Estudiante").checked).toBeFalsy();
expect(queryByLabelText("Profesor").checked).toBeTruthy();
expect(queryByLabelText("Administrador").checked).toBeFalsy();

fireEvent.click(getByLabelText("Administrador"));

expect(queryByLabelText("Estudiante").checked).toBeFalsy();
expect(queryByLabelText("Profesor").checked).toBeFalsy();
expect(queryByLabelText("Administrador").checked).toBeTruthy();
});
