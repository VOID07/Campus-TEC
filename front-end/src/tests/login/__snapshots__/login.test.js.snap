// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verificar la ausencia de cambios inesperados en el GUI 1`] = `
<div
  className="App container"
>
  <div
    className="container"
  >
    <nav
      className="NavBarTop navbar navbar-expand navbar-light fixed-top"
    >
      <a
        className="navbar-brand"
        href="#"
      >
        <div>
          <img
            alt="logo"
            className="logoEdit"
            src="logo.svg"
          />
        </div>
      </a>
    </nav>
  </div>
  <div
    className="container"
  >
    <div
      className="Card card"
    >
      <div
        className="Logo"
      >
        <img
          className="card-img-top"
          src="logo.svg"
        />
        <div
          style={
            Object {
              "margin": "20px",
            }
          }
        >
          <h4>
            INGRESAR
          </h4>
        </div>
      </div>
      <div
        className="CardBody card-body"
      >
        <form
          className=""
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="formBasicEmail"
            >
              Usuario
            </label>
            <input
              className="form-control"
              id="formBasicEmail"
              onChange={[Function]}
              placeholder="Usuario TEC"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="formBasicPassword"
            >
              Contrase単a
            </label>
            <input
              className="form-control"
              id="formBasicPassword"
              onChange={[Function]}
              placeholder="Contrase単a"
              type="password"
              value=""
            />
          </div>
          <div>
            <div
              className="checkBoxes form-group"
            >
              <div
                className="form-check"
              >
                <input
                  checked={true}
                  className="form-check-input"
                  disabled={false}
                  id="checkbox1"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="form-check-label"
                  htmlFor="checkbox1"
                  title=""
                >
                  Estudiante
                </label>
              </div>
              <div
                className="form-check"
              >
                <input
                  checked={false}
                  className="form-check-input"
                  disabled={false}
                  id="checkbox2"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="form-check-label"
                  htmlFor="checkbox2"
                  title=""
                >
                  Profesor
                </label>
              </div>
              <div
                className="form-check"
              >
                <input
                  checked={false}
                  className="form-check-input"
                  disabled={false}
                  id="checkbox3"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="form-check-label"
                  htmlFor="checkbox3"
                  title=""
                >
                  Administrador
                </label>
              </div>
            </div>
          </div>
          <div
            className="d-flex justify-content-center"
          >
            <button
              className="loginBtn btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              ENTRAR
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <nav
      className="NavBarBot navbar navbar-expand navbar-light bg-#153c6c fixed-bottom"
    />
  </div>
</div>
`;

exports[`Verificar la ausencia de cambios inesperados en el GUI 2`] = `
<div
  className="App container"
>
  <div
    className="container"
  >
    <nav
      className="NavBarTop navbar navbar-expand navbar-light fixed-top"
    >
      <a
        className="navbar-brand"
        href="#"
      >
        <div>
          <img
            alt="logo"
            className="logoEdit"
            src="logo.svg"
          />
        </div>
      </a>
    </nav>
  </div>
  <div
    className="container"
  >
    <div
      className="Card card"
    >
      <div
        className="Logo"
      >
        <img
          className="card-img-top"
          src="logo.svg"
        />
        <div
          style={
            Object {
              "margin": "20px",
            }
          }
        >
          <h4>
            INGRESAR
          </h4>
        </div>
      </div>
      <div
        className="CardBody card-body"
      >
        <form
          className=""
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="formBasicEmail"
            >
              Usuario
            </label>
            <input
              className="form-control"
              id="formBasicEmail"
              onChange={[Function]}
              placeholder="Usuario TEC"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="formBasicPassword"
            >
              Contrase単a
            </label>
            <input
              className="form-control"
              id="formBasicPassword"
              onChange={[Function]}
              placeholder="Contrase単a"
              type="password"
              value=""
            />
          </div>
          <div>
            <div
              className="checkBoxes form-group"
            >
              <div
                className="form-check"
              >
                <input
                  checked={false}
                  className="form-check-input"
                  disabled={false}
                  id="checkbox1"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="form-check-label"
                  htmlFor="checkbox1"
                  title=""
                >
                  Estudiante
                </label>
              </div>
              <div
                className="form-check"
              >
                <input
                  checked={true}
                  className="form-check-input"
                  disabled={false}
                  id="checkbox2"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="form-check-label"
                  htmlFor="checkbox2"
                  title=""
                >
                  Profesor
                </label>
              </div>
              <div
                className="form-check"
              >
                <input
                  checked={false}
                  className="form-check-input"
                  disabled={false}
                  id="checkbox3"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="form-check-label"
                  htmlFor="checkbox3"
                  title=""
                >
                  Administrador
                </label>
              </div>
            </div>
          </div>
          <div
            className="d-flex justify-content-center"
          >
            <button
              className="loginBtn btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              ENTRAR
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <nav
      className="NavBarBot navbar navbar-expand navbar-light bg-#153c6c fixed-bottom"
    />
  </div>
</div>
`;

exports[`Verificar la ausencia de cambios inesperados en el GUI 3`] = `
<div
  className="App container"
>
  <div
    className="container"
  >
    <nav
      className="NavBarTop navbar navbar-expand navbar-light fixed-top"
    >
      <a
        className="navbar-brand"
        href="#"
      >
        <div>
          <img
            alt="logo"
            className="logoEdit"
            src="logo.svg"
          />
        </div>
      </a>
    </nav>
  </div>
  <div
    className="container"
  >
    <div
      className="Card card"
    >
      <div
        className="Logo"
      >
        <img
          className="card-img-top"
          src="logo.svg"
        />
        <div
          style={
            Object {
              "margin": "20px",
            }
          }
        >
          <h4>
            INGRESAR
          </h4>
        </div>
      </div>
      <div
        className="CardBody card-body"
      >
        <form
          className=""
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="formBasicEmail"
            >
              Usuario
            </label>
            <input
              className="form-control"
              id="formBasicEmail"
              onChange={[Function]}
              placeholder="Usuario TEC"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="formBasicPassword"
            >
              Contrase単a
            </label>
            <input
              className="form-control"
              id="formBasicPassword"
              onChange={[Function]}
              placeholder="Contrase単a"
              type="password"
              value=""
            />
          </div>
          <div>
            <div
              className="checkBoxes form-group"
            >
              <div
                className="form-check"
              >
                <input
                  checked={false}
                  className="form-check-input"
                  disabled={false}
                  id="checkbox1"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="form-check-label"
                  htmlFor="checkbox1"
                  title=""
                >
                  Estudiante
                </label>
              </div>
              <div
                className="form-check"
              >
                <input
                  checked={false}
                  className="form-check-input"
                  disabled={false}
                  id="checkbox2"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="form-check-label"
                  htmlFor="checkbox2"
                  title=""
                >
                  Profesor
                </label>
              </div>
              <div
                className="form-check"
              >
                <input
                  checked={true}
                  className="form-check-input"
                  disabled={false}
                  id="checkbox3"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="form-check-label"
                  htmlFor="checkbox3"
                  title=""
                >
                  Administrador
                </label>
              </div>
            </div>
          </div>
          <div
            className="d-flex justify-content-center"
          >
            <button
              className="loginBtn btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              ENTRAR
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <nav
      className="NavBarBot navbar navbar-expand navbar-light bg-#153c6c fixed-bottom"
    />
  </div>
</div>
`;

exports[`Verificar la ausencia de cambios inesperados en el GUI 4`] = `
<div
  className="App container"
>
  <div
    className="container"
  >
    <nav
      className="NavBarTop navbar navbar-expand navbar-light fixed-top"
    >
      <a
        className="navbar-brand"
        href="#"
      >
        <div>
          <img
            alt="logo"
            className="logoEdit"
            src="logo.svg"
          />
        </div>
      </a>
    </nav>
  </div>
  <div
    className="container"
  >
    <div
      className="Card card"
    >
      <div
        className="Logo"
      >
        <img
          className="card-img-top"
          src="logo.svg"
        />
        <div
          style={
            Object {
              "margin": "20px",
            }
          }
        >
          <h4>
            INGRESAR
          </h4>
        </div>
      </div>
      <div
        className="CardBody card-body"
      >
        <form
          className=""
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="formBasicEmail"
            >
              Usuario
            </label>
            <input
              className="form-control"
              id="formBasicEmail"
              onChange={[Function]}
              placeholder="Usuario TEC"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-label"
              htmlFor="formBasicPassword"
            >
              Contrase単a
            </label>
            <input
              className="form-control"
              id="formBasicPassword"
              onChange={[Function]}
              placeholder="Contrase単a"
              type="password"
              value=""
            />
          </div>
          <div>
            <div
              className="checkBoxes form-group"
            >
              <div
                className="form-check"
              >
                <input
                  checked={true}
                  className="form-check-input"
                  disabled={false}
                  id="checkbox1"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="form-check-label"
                  htmlFor="checkbox1"
                  title=""
                >
                  Estudiante
                </label>
              </div>
              <div
                className="form-check"
              >
                <input
                  checked={false}
                  className="form-check-input"
                  disabled={false}
                  id="checkbox2"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="form-check-label"
                  htmlFor="checkbox2"
                  title=""
                >
                  Profesor
                </label>
              </div>
              <div
                className="form-check"
              >
                <input
                  checked={false}
                  className="form-check-input"
                  disabled={false}
                  id="checkbox3"
                  onChange={[Function]}
                  type="checkbox"
                />
                <label
                  className="form-check-label"
                  htmlFor="checkbox3"
                  title=""
                >
                  Administrador
                </label>
              </div>
            </div>
          </div>
          <div
            className="d-flex justify-content-center"
          >
            <button
              className="loginBtn btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              ENTRAR
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <nav
      className="NavBarBot navbar navbar-expand navbar-light bg-#153c6c fixed-bottom"
    />
  </div>
</div>
`;
